From: =?utf-8?q?Lukas_M=C3=A4rdian?= <lukas.maerdian@oss.qualcomm.com>
Date: Tue, 22 Sep 2026 00:00:00 +0200
Subject: build: add PAL functional test program

Build the existing PAL ultrasound use-case test as a Linux executable.
The Debian package installs it separately so it can be run on target
hardware without adding it to the normal PAL runtime package.

This is intentionally not registered as an Automake TESTS target: it
requires a functioning AudioReach sound card and target configuration.

Forwarded: not-needed
---
 Makefile.am           |  2 +-
 configure.ac          |  1 +
 test/Makefile.am      | 11 +++++++++++
 test/PalTest_main.c   |  1 +
 test/PalUsecaseTest.c |  2 ++
 5 files changed, 16 insertions(+), 1 deletion(-)
 create mode 100644 test/Makefile.am

diff --git a/Makefile.am b/Makefile.am
index 69a601e..fa0ed70 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -179,4 +179,4 @@ audioadsprpcd_la_CFLAGS = -fPIC
 endif
 
 
-SUBDIRS =  inc plugins . session stream device plugins/configs
+SUBDIRS =  inc plugins . session stream device plugins/configs test
diff --git a/configure.ac b/configure.ac
index 7a285a2..b3fbf7e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -173,6 +173,7 @@ AC_CONFIG_SUBDIRS([
 ])
 
 AC_CONFIG_FILES([ Makefile \
+test/Makefile \
 pal.pc \
 stream/Makefile \
 session/Makefile \
diff --git a/test/Makefile.am b/test/Makefile.am
new file mode 100644
index 0000000..e1c1abd
--- /dev/null
+++ b/test/Makefile.am
@@ -0,0 +1,11 @@
+bin_PROGRAMS = PalTest
+
+PalTest_SOURCES = \
+    PalUsecaseTest.c \
+    PalTest_main.c
+
+PalTest_CPPFLAGS = \
+    -I$(top_srcdir)/inc
+
+PalTest_LDADD = $(top_builddir)/.libs/libaudioreach-pal.so
+PalTest_DEPENDENCIES = $(top_builddir)/libaudioreach-pal.la
diff --git a/test/PalTest_main.c b/test/PalTest_main.c
index 1f8ec17..049ac36 100644
--- a/test/PalTest_main.c
+++ b/test/PalTest_main.c
@@ -33,6 +33,7 @@
  */
 
 #include <unistd.h>
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <signal.h>
diff --git a/test/PalUsecaseTest.c b/test/PalUsecaseTest.c
index b69ca37..216f611 100644
--- a/test/PalUsecaseTest.c
+++ b/test/PalUsecaseTest.c
@@ -29,6 +29,8 @@
 
 #include"PalUsecaseTest.h"
 #include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 
 static struct pal_stream_attributes *stream_attributes;
